packages/agents/src/mixin.ts

```diff
@@ -0,0 +1,648 @@
+    const parsedEnv = aiGatewayEnvSchema.safeParse(c.env);
+    if (!parsedEnv.success) {
+      console.error("Invalid environment variables:", parsedEnv.error);
+      return c.json(
+        {
+          error: "Missing or invalid environment variables",
+          details: parsedEnv.error.format(),
+        },
+        422,
+      );
+    }
+
+    const client = new Cloudflare({
+      apiToken: parsedEnv.data.CLOUDFLARE_API_TOKEN,
+    });
+
+    const gateways = await client.aiGateway.list({
+      account_id: parsedEnv.data.CLOUDFLARE_ACCOUNT_ID,
+    });
+
+    if (!gateways) {
+      return c.json({ error: "Failed to retrieve gateways" }, 500);
+    }
+    return c.json(gateways.result);
+  });
+
+  // Get log list for a specific AI gateway
+  router.get(
+    "/agents/:namespace/:instance/admin/ai-gateways/:id/logs",
+    async (c) => {
+      // Extract required environment variables
+
+      const parsedEnv = aiGatewayEnvSchema.safeParse(c.env);
+      if (!parsedEnv.success) {
+        console.error("Invalid environment variables:", parsedEnv.error);
+        return c.json(
+          {
+            error: "Missing or invalid environment variables",
+            details: parsedEnv.error.format(),
+          },
+          422,
+        );
+      }
+
+      const client = new Cloudflare({
+        apiToken: parsedEnv.data.CLOUDFLARE_API_TOKEN,
+      });
+      const { id } = c.req.param();
+
+      const logs = await client.aiGateway.logs.list(id, {
+        account_id: parsedEnv.data.CLOUDFLARE_ACCOUNT_ID,
+      });
+
+      if (!logs) {
+        return c.json({ error: "Failed to retrieve logs" }, 500);
+      }
+
+      return c.json(logs.result);
+    },
+  );
+
+  // Get log details for a specific gateway/log
+  router.get(
+    "/agents/:namespace/:instance/admin/ai-gateways/:id/logs/:logId",
+    async (c) => {
+      // Extract required environment variables
+      const parsedEnv = aiGatewayEnvSchema.safeParse(c.env);
+      if (!parsedEnv.success) {
+        console.error("Invalid environment variables:", parsedEnv.error);
+        return c.json(
+          {
+            error: "Missing or invalid environment variables",
+            details: parsedEnv.error.format(),
+          },
+          422,
+        );
+      }
+
+      const client = new Cloudflare({
+        apiToken: parsedEnv.data.CLOUDFLARE_API_TOKEN,
+      });
+      const { id, logId } = c.req.param();
+
+      const log = await client.aiGateway.logs.get(id, logId, {
+        account_id: parsedEnv.data.CLOUDFLARE_ACCOUNT_ID,
+      });
+
+      if (!log) {
+        return c.json({ error: "Failed to retrieve log" }, 500);
+      }
+
+      return c.json(log);
+    },
+  );
+
+  return router;
+}
+
+interface ObservedProperties {
+  _activeStreams: Set<SSEStreamingApi>;
+  _mcpConnections?: Map<string, MCPClientConnection>;
+  _fiberRouter?: Hono;
+}
+
+/**
+ * Mixin factory that adds Fiber capabilities to Agent classes
```

Comment:
can we have a better comment here.

maybe:

"Mixin factory that adds observability capabilities to Agent classes allowing them to be viewed in the Fiberplane agents playground"